packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -146,16 +149,19 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
     _setTween();
   }
 
-  void _handleTapDown(PointerDownEvent event) {
-    _animationController.animateTo(1.0, duration: kFadeOutDuration);
+  Future<Null> _handleTapDown(PointerDownEvent event) async {
+    await _lastAnimation;
+    _lastAnimation = _animationController.animateTo(1.0, duration: kFadeOutDuration);
```

Comment:
Refactored to ensure that we don't chain fade-in/fade-out animation sets. Also added the "mounted" check.

Furthermore, I tweaked the animation timing a little bit; for me, it looks closer to what iOS is doing and it also makes it much harder (impossible?) to trigger the edge cases where you get a down and up event while the previous up animation is still in progress.

The "fix" to use onTapDown/onTapUp/onTapCancel from a GestureDetector doesn't work well as it delays the firing of onTapDown by 100ms (`kPressTimeout`), which makes the button seem unresponsive.